pkg/cmd/secvulnMaven.go

```diff
@@ -5,33 +5,319 @@
+	}
+
+	updateParent()
+	fmt.Printf("Security scanning project pom.xml updated\n")
+
+}
+
+func startScanningPipeline(mainPomUrl string) {
+
+	fmt.Printf("Starting the scanning pipeline at %v\n", mainPomUrl)
+
+	mainPom, err := readPomFromUrl(mainPomUrl)
+	if err != nil {
+		fmt.Printf("Unable to find the main pom at %v\n", mainPomUrl)
+		panic(err)
+	}
+
+	createPseudoMavenProject(mainPom)
+	fmt.Printf("Pseudo maven project created for: %v\n", mainPom.ArtifactId)
+
+	// Repeat the process for all projects in this dependency chain if the groupId is dev.galasa
+	fmt.Printf("Pseudo maven projects created for dependency chain of %v:\n", mainPom.ArtifactId)
+
+	for _, project := range toDoProjects {
+
+		// If this project has already been processed then continue
+		if checkIfCompleted(project) {
+			continue
+		}
+
+		// Get the groupId of this project from the main pom to use in the url
+		var groupId string
+		if groupId = getGroupId(project, mainPom); groupId == "" {
+			fmt.Printf("Unable to get groupId for artifact %s\n", project)
+			panic(nil)
+		}
+		// Get the version of this project from the main pom to use in the url
+		var version string
+		if version = getVersion(project, mainPom); version == "" {
+			fmt.Printf("Unable to get version for artifact %s\n", project)
+			panic(nil)
+		}
+
+		// Get the current pom for this project to use to create the stripped down pom for the pseudo maven project
+		var currentPom Pom
+		if project == "com.jcraft.jsch" {
+			// com.jcraft.jsch is an exception to the normal format
+			currentPom, err = readPomFromRepos("jsch", "com.jcraft", version)
+		} else {
+			currentPom, err = readPomFromRepos(project, groupId, version)
+		}
+		if err != nil || (currentPom.ArtifactId != project && currentPom.ArtifactId != "jsch") {
+			fmt.Printf("Could not find pom for artifact %s\n", project)
+			panic(err)
+		}
+
+		createPseudoMavenProject(currentPom)
+
+		fmt.Printf("- %s\n", project)
+
+	}
+
+}
+
+func readPomFromUrl(url string) (Pom, error) {
+
+	var pom Pom
+
+	response, err := http.Get(url)
+	if err != nil {
+		return pom, err
+	}
+
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return pom, err
+	}
+
+	err = xml.Unmarshal(body, &pom)
+	if err != nil {
+		return pom, err
+	}
+
+	return pom, nil
+
+}
+
+func readPomFromRepos(artifactName, groupId, version string) (Pom, error) {
+
+	var pom Pom
+	var err error
+
+	var url string
+	// Iterate through the provided repos with the --repo tag
+	for _, repo := range *secvulnMavenPomRepos {
+		// Use the groupId, artifactName and version to build up the url of the pom
+		s := strings.Split(groupId, ".")
+		var group string
+		for _, e := range s {
+			group += e + "/"
+		}
+
+		url = fmt.Sprintf("%s/%s%s/%s/%s-%s.pom", repo, group, artifactName, version, artifactName, version)
+
+		pom, err = readPomFromUrl(url)
+		if pom.ArtifactId == artifactName {
+			return pom, nil
+		}
+	}
+
+	return pom, err
```

Comment:
what happens if the the artifact is not found in any repo?

Reply:
it would be better if you returned a custom error.

return nil,errors.New("Project not located in repos")

Makes much clear to the person reading the code, and explicitly sets an error instead of assuming it will be set ealier in the code.   If someone changes the code above,   then it may have unintended consequences